test/integration/targets/aws_ses_rule_set/tasks/main.yaml

```diff
@@ -0,0 +1,451 @@
+---
+- name: set up aws connection info
+  set_fact:
+    aws_connection_info: &aws_connection_info
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+# ============================================================
+- name: test create rule sets
+  block:
+    - name: create rule set
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        <<: *aws_connection_info
+      register: result
+    - name: assert changed to exists inactive
+      assert:
+        that:
+          - result.changed == True
+          - result.active == False
+          - "default_rule_set in result.rule_sets|map(attribute='name')"
+    - name: recreate rule set
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        <<: *aws_connection_info
+      register: result
+    - name: assert changed is False
+      assert:
+        that:
+          - result.changed == False
+  always:
+    - name: cleanup rule set
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        state: absent
+        force: True
+        <<: *aws_connection_info
+# ============================================================
+- name: mark rule set active
+  block:
+    - name: create rule set
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        <<: *aws_connection_info
+    - name: mark rule set active
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        active: True
+        <<: *aws_connection_info
+      register: result
+    - name: assert changed to active
+      assert:
+        that:
+          - result.changed == True
+          - result.active == True
+    - name: remark rule set active
+      aws_ses_rule_set:
+        name: "{{ default_rule_set }}"
+        active: True
+        <<: *aws_connection_info
+      register: result
+    - name: assert changed is False
+      assert:
+        that:
+          - result.changed == False
```

Comment:
This seems to be periodically failing with changed is True. I think the module may need a waiter.

Reply:
I might be wrong, but I don't think this is a timing / eventual consistency thing.

There can only be one active rule set per-region per-account. I think what's happening is the python2 and python3 tests are running in parallel and if we're unlucky, the other test creates an active rule in the gap here and the test fails becuse this rule is no-longer active so making it active is a change.

Hard to be sure because it's intermittent, but I've repeatedly run ths test on an ec2 instance in the same region as the SES instance and not had a single failure, but when I run 2 instances in parallel of just this test it fails reliably as described.

I'll have to give some thought to how to have a reliable test here since the active ruleset really is global state.

Any advice or thoughts would be appreciated.